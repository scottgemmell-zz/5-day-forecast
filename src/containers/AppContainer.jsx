import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
//import { getForecastData } from "../redux/actions";
import { Grid } from "react-bootstrap";
import Controls from "../components/Controls.jsx";
import Forecasts from "../components/Forecasts.jsx";

class AppContainer extends Component {
	render() {
		const { data } = this.props;

		return (
			<div className="app">
				<Controls />
				<Grid>
					{(data !== "") && <Forecasts />}
				</Grid>
			</div>
		);
	}
}

AppContainer.propTypes = {
	data: PropTypes.object,
};

const mapStateToProps = (state) => ({
	data: state.forecasts,
});

export default connect(
	mapStateToProps
)(AppContainer);
