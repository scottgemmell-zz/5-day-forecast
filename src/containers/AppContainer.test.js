import React from "react";
import AppContainer from "./AppContainer";
import { shallow } from "enzyme";

const defaultProps = () => ({});
const setup = (overrideProps = {}) => {
    const props = Object.assign({}, defaultProps(), overrideProps);
    const wrapper = shallow(<AppContainer {...props} />);
    return { wrapper, props };
};

describe("<AppContainer />", () => {
    it("renders", () => {
        const { wrapper } = setup();
        expect(wrapper).toMatchSnapshot();
    });
});