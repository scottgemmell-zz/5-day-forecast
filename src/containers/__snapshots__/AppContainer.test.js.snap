// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppContainer /> renders 1`] = `ShallowWrapper {}`;
